src/Data/Yaml.hs

```diff
@@ -71,11 +71,11 @@ module Data.Yaml
       -- * Custom encoding
     , objToEvents
     , isSpecialString
-    , EncodeOptions
+    , EncodeOptions(..)
```

Comment:
This data type is intentional opaque, please undo this change.

Reply:
I figured, but one would need this exported to implement an `encode`-like functionality.

E.g. this is how I implemented the encoding of many documents using this PR:
```haskell
-- | Encode a list of documents into their YAML representation,
--   interspersed with a separator
encodeMany :: Data.Yaml.ToJSON a => [a] -> IO ByteString
encodeMany objs = Data.Conduit.runConduitRes
    $ Data.Conduit.List.sourceList events
        .| Text.Libyaml.encodeWith (Data.Yaml.encodeOptionsFormat opts)
  where
    opts = Data.Yaml.defaultEncodeOptions

    events = concat
      [ [ Text.Libyaml.EventStreamStart ]
      , encodeDocument =<< objs
      , [ Text.Libyaml.EventStreamEnd ]
      ]

    encodeDocument obj
      = filter (\el -> (el /= Text.Libyaml.EventStreamStart)
                    && (el /= Text.Libyaml.EventStreamEnd))
      $ Data.Yaml.objToEvents opts
      $ Data.Yaml.toJSON obj
```

As I said above, if instead we'd like to not expose this stuff and add a switch (in `EncodeOptions`) that you can flip to get many documents instead, I'd be fine implementing that as well

Reply:
Right sorry, I'll add more context and clarify my use-case:
- Use-case: "I'd like to output several yaml documents in the same stream".
  Documents in the same yaml stream are divided by a "separator":
  ```yaml
  1
  ---
  2
  ```
- Problem: this is not currently supported in this library, since the fact that we output only one document is hardcoded [here in `objToEvents`](https://github.com/snoyberg/yaml/blob/master/src/Data/Yaml.hs#L200)
- Possible solution: expose all the facilities that `encodeWith` uses to go from `Value` to `ByteString`, so that a user can implement their own versions of it (as I did with `encodeMany` above - it takes a `[Value]`, because the list here represents the list of documents).
  In this case this means exposing `objToEvents` and the `EncodeOptions` constructors
- Possibly better solution: add a field to `EncodeOptions`, something like `encodeOptionsEventsFn :: EncodeOptions -> Value -> [Text.Libyaml.Event]`.
  Default for this setting would be `objToEvents`.
  We should expose `objToEvents` also in this case, so that a user can wrap it to output the necessary events (as I did up there in `encodeMany`).
  This is "the switch" that I mentioned in the previous comment. Sorry for not making this clear in the body of the PR, I should have mentioned there that I meant to add a setting.
- Another possible solution: make "multiple documents" just the default behaviour and have `encodeWith` take `[Value]` instead of `Value`, where the list represents the list of documents to encode (and this should answer the question of "why would we need to pass a `[Value]` instead of `Value`")

Does it make sense?

Reply:
I do not understand what `encodeOptionsEventsFn` would do. I also don't understand why exposing the `EncodeOptions` constructor and fields is necessary for this. I'm still quite confused by what's being asked for here. I'd strongly suggest starting over again, with a new issue, clarifying better what the goals are here. From my perspective:

* There's already raw access to do what you want with the Text.Libyaml module
* There's some logic in Data.Yaml you want access to in order to ease the task
* I don't understand why you need access to the things you're asking for access to